aws_doc_sdk_examples_tools/doc_gen.py

```diff
@@ -123,34 +123,46 @@ def clone(self) -> "DocGen":
 
     def for_root(self, root: Path, config: Optional[Path] = None) -> "DocGen":
         self.root = root
-        metadata = root / ".doc_gen/metadata"
 
         if config is None:
             config = Path(__file__).parent / "config"
 
-        with open(root / ".doc_gen" / "validation.yaml") as file:
-            validation = yaml.safe_load(file)
-            self.validation.allow_list.update(validation.get("allow_list", []))
-            self.validation.sample_files.update(validation.get("sample_files", []))
-
-        with open(config / "sdks.yaml", encoding="utf-8") as file:
-            meta = yaml.safe_load(file)
-            sdks, errs = parse_sdks("sdks.yaml", meta)
-            self.errors.extend(errs)
-
-        with open(config / "services.yaml", encoding="utf-8") as file:
-            meta = yaml.safe_load(file)
-            services, service_errors = parse_services("services.yaml", meta)
-            self.errors.extend(service_errors)
-
-        cross = set(
-            [path.name for path in (metadata.parent / "cross-content").glob("*.xml")]
-        )
+        try:
+            with open(root / ".doc_gen" / "validation.yaml") as file:
+                validation = yaml.safe_load(file)
+                self.validation.allow_list.update(validation.get("allow_list", []))
+                self.validation.sample_files.update(validation.get("sample_files", []))
+        except Exception:
+            pass
+
+        try:
+            with open(config / "sdks.yaml", encoding="utf-8") as file:
+                meta = yaml.safe_load(file)
+                sdks, errs = parse_sdks("sdks.yaml", meta)
+                self.sdks = sdks
+                self.errors.extend(errs)
+        except Exception:
+            pass
+
+        try:
+            with open(config / "services.yaml", encoding="utf-8") as file:
+                meta = yaml.safe_load(file)
+                services, service_errors = parse_services("services.yaml", meta)
+                self.services = services
+                self.errors.extend(service_errors)
+        except Exception:
+            pass
```

Comment:
If these silently fail then you'll have empty sdks and services, which will blow up later. Better to error out here.

Reply:
That's the default state, and there are clear error messages then. In fact, most tributaries are not expected to have sdks or services, so they need to be empty and need to inherit from the parent doc_gen.

Reply:
If these files don't exist at all, even in the root repo, the first place I see an error happening is in metadata.parse_services, where it will raise an UnknownService exception. This seems a lot harder to track down than "no services defined" or something in this doc_gen function. I see that you don't want to error out on a missing file, but I think you'll want to check that at least some sdks and services exist before you start the parse_examples loop.

Reply:
It's not an error here until the consumer of for_root says it is. doc_gen.py doesn't care if there are services or sdks or validation. Only consumers do.

Reply:
But it is an error here if there are no services loaded at all. In this function it eventually calls parse_examples, which ultimately calls metadata.parse_services, which will fail with UnknownService. I think this is a bad experience and easily overcome by checking that sdks and services are not None or empty. I see that this is unlikely in that the main root would have to omit one of these files, so arguably "it will never happen", although saying that means it is now guaranteed to happen. If we just need to get this in today that's fine, I don't want to block on this.